tests/pytest.ini

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[pytest]
+addopts =
+    -rasl
+    --cov=airflow/
+    --cov-report html:airflow/www/static/coverage/
+    --ignore=tests/dags/test_dag_serialization.py
+;    This will treat all tests as flaky
+;    --force-flaky
+norecursedirs =
+    tests/dags_with_system_exit
+    tests/test_utils
+    tests/dags_corrupted
+faulthandler_timeout=180
+log_print = True
+log_level = INFO
+filterwarnings =
+    ignore::DeprecationWarning
+    ignore::PendingDeprecationWarning
+    ignore::RuntimeWarning
```

Comment:
I'm not sure about these, but especially this one.

What was your thinking for this?

Reply:
The general idea was to declutter logs by removing deprecation warnings. I have to check why I added the `RuntimeWarning` but I can agree that this one is quite disputable.

Reply:
I see no `RuntimeWarning` now. What do you think about filtering out other warnings? @potiuk @mik-laj @ashb @feluelle

Reply:
I am all ok for filtering warnings but only when we run in CI. If we run tests locally, we should see the warnings. I think there is very little value in warnings displayed in CI for regular PRs. Usually those will be the "other" warnings that you are not really interested in. 

Maybe, what could be done instead - we have the CRON build run daily where we already have a more comprehensive tests (for example we build the whole Docker image from the scratch and we always run Kubernetes tests no matter if any of kubernetes files changed). Maybe we could change it that Deprecation warnings are still displayed when you run tests locally and when you run CRON CI job, but not when you run regular PR.

Reply:
I think we should differ between `DeprecationWarning`s in our Airflow code base and external modules. IMO we should always try to fix deprecation warnings in our code base. And we can ignore the external ones.

Reply:
> And we can ignore the external ones.

Because sometimes those `DeprecationWarnings` of external modules cannot be fixed by us.

Reply:
I agree with both of you @feluelle and @potiuk. Removing depreciation warning adds around 800 lines to CI log. So I would do as @potiuk suggested: filter warning only when run on CI.

Reply:
I am fine with that. :) 👍